src/clj/puppetlabs/trapperkeeper/services/protocols/filesystem_watch_service.clj

```diff
@@ -6,9 +6,15 @@
   "Schema for an event on a file watched by this service."
   (schema/if #(= (:type %) :unknown)
     {:type (schema/eq :unknown)
-     :path (schema/pred nil?)}
+     :count schema/Int
+     :watched-path File
+     :changed-path (schema/pred nil?)
+     :full-path (schema/pred nil?)}
```

Comment:
I just mentioned this on one of the previous PRs in this repo but I'll restate it here for convenience ...

`(schema/pred nil?)` looks very strange to me, and I think `schema/optional-key` is far more conventional.